### DISCUSSION TITLE
Tree-sitter-graph

### DISCUSSION THREAD

**Discussion by tinygiant98** [tree-sitter/tree-sitter#1568] [Q&A] [answered]
Wondering if anyone has seen any practical implementation examples for tree-sitter-graph.  I know it's a separate repo, but I'm hoping, given the lack of discussion options there, that there may be some cross-development knowledge here.  I was able to get some local, volatile results from tree-sitter-graph executable, but looking for best practices/methodologies for integrating into a language-server style setup.

**Comment by dcreager** [comment, marked as answer]
Hi there! Unfortunately tree-sitter-graph is still in its early stages. The [language spec](https://docs.rs/tree-sitter-graph/*/tree_sitter_graph/reference/index.html) is approaching stability, though we're still thinking through whether we wany to move to a lazy execution model.  We have an internal predecessor to that graph DSL running internally producing the data for the new [precise code nav](https://github.blog/2021-12-09-precise-code-navigation-python-code-navigation-pull-requests/) data on GitHub for Python.  Over the next couple of weeks, we'll be finishing up the "spackle" work that will connect the tree-sitter-graph and [stack-graphs](https://docs.rs/stack-graphs/*/stack_graphs/) crates, so that we can produce the code nav data using them.  As part of that, we'll try to make sure that there's reasonable documentation and examples that show how others can use them too.

**Reply by tinygiant98** [reply]
Ok, thanks for the info. Good luck, looks like an amazing concept.
